### PR TITLE
Add CopyToRepoWithBackupRepository and CopyFromRepoWithBackupRepository in data_mover

### PR DESCRIPTION
**Pull request by xinyanw409**
Update CopyToRepo and CopyFromRepo in data mover with BackupRepository to get S3PETM from BackupRepository params instead of hard coding S3PETM when initializing data mover.

Precheck(ongoing):
https://container-dp.svc.eng.vmware.com/job/Container_Precheck_Velero/217/

### CODE REVIEW

**Comment by svcbot-qecnsdp** [comment]
Congratulations !! 

All the pre-check tests passed
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/45/

File: pkg/dataMover/data_mover.go [hunk 1]
```diff
@@ -101,6 +112,13 @@ func (this *DataMover) CopyToRepo(peID astrolabe.ProtectedEntityID) (astrolabe.P
 	this.RegisterOngoingUpload(peID, cancelFunc)
 
 	log.Debugf("Ready to call s3 PETM copy API for local PE")
+	if backupRepository.Name != "Not use" {
```
Comment (deepakkinni) [inline pkg/dataMover/data_mover.go:115]: can we create a constant, and use it here?, something like "not-applicable"

File: pkg/dataMover/data_mover.go [hunk 2]
```diff
@@ -101,6 +112,13 @@ func (this *DataMover) CopyToRepo(peID astrolabe.ProtectedEntityID) (astrolabe.P
...
+		s3PETM, err := utils.GetRepositoryFromBackupRepository(backupRepository, log)
```
Comment (deepakkinni) [inline pkg/dataMover/data_mover.go:116]: Although `GetRepositoryFromBackupRepository` is not part of this change, would it be possible to rename it to `GetS3RepositoryFromBackupRepository`, I think it would help anyone reading the code.

File: pkg/dataMover/data_mover.go [hunk 3]
```diff
@@ -101,6 +112,13 @@ func (this *DataMover) CopyToRepo(peID astrolabe.ProtectedEntityID) (astrolabe.P
...
+		if err != nil {
+			log.Errorf("Failed to get S3 repository from backup repository %s", backupRepository.Name)
```
Comment (dsu-igeek) [inline pkg/dataMover/data_mover.go:118]: Why is this continuing on?  This should be returning the err here, the s3PETM is not valid

File: pkg/dataMover/data_mover.go (hunk 2 above)
Comment (dsu-igeek) [reply pkg/dataMover/data_mover.go:116]: Let's change s3PETM to repoPETM.  We may use something besides S3 here

File: pkg/dataMover/data_mover.go [hunk 4]
```diff
@@ -113,8 +131,25 @@ func (this *DataMover) CopyToRepo(peID astrolabe.ProtectedEntityID) (astrolabe.P
 }
 
 func (this *DataMover) CopyFromRepo(peID astrolabe.ProtectedEntityID) (astrolabe.ProtectedEntityID, error) {
+	backupRepository := builder.ForBackupRepository("Not use").Result()
+	return this.copyFromRepo(peID, backupRepository)
+}
+
+func (this *DataMover) CopyFromRepoWithBackupRepository(peID astrolabe.ProtectedEntityID, backupRepository *backupdriverv1.BackupRepository) (astrolabe.ProtectedEntityID, error) {
+	this.Info("CopyFromRepoWithBackupRepository was called")
```
Comment (dsu-igeek) [inline pkg/dataMover/data_mover.go:139]: Let's be consistent.  Everywhere else we are using log.Info

File: pkg/dataMover/data_mover.go [hunk 5]
```diff
@@ -113,8 +131,25 @@ func (this *DataMover) CopyToRepo(peID astrolabe.ProtectedEntityID) (astrolabe.P
...
+	return this.copyFromRepo(peID, backupRepository)
+}
+
+func (this *DataMover) copyFromRepo(peID astrolabe.ProtectedEntityID, backupRepository *backupdriverv1.BackupRepository) (astrolabe.ProtectedEntityID, error) {
 	log := this.WithField("Remote PEID", peID.String())
 	log.Infof("Copying the snapshot from remote repository to local.")
+	if backupRepository.Name != "Not use" {
```
Comment (dsu-igeek) [inline pkg/dataMover/data_mover.go:146]: If Name is set to "Not use', what will the setting for S3PETM be?  How will we avoid a NPE at line 119?

File: pkg/dataMover/data_mover.go (hunk 1 above)
Comment (dsu-igeek) [reply pkg/dataMover/data_mover.go:115]: If Name is "Not use" how do we avoid an NPE at line 104

File: pkg/dataMover/data_mover.go (hunk 2 above)
Comment (xinyanw409) [reply pkg/dataMover/data_mover.go]: > Although `GetRepositoryFromBackupRepository` is not part of this change, would it be possible to rename it to `GetS3RepositoryFromBackupRepository`, I think it would help anyone reading the code.

As we may use some other repo in the future, so I think the general name GetRepositoryFromBackupRepository may be better.

File: pkg/dataMover/data_mover.go (hunk 1 above)
Comment (xinyanw409) [reply pkg/dataMover/data_mover.go]: > If Name is "Not use" how do we avoid an NPE at line 104

line 104 is "updatedPE, err := this.ivdPETM.GetProtectedEntity(ctx, peID)", this will not cause an NPE.

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/51/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/55/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/63/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/68/

File: pkg/dataMover/data_mover.go [hunk 6]
```diff
@@ -87,6 +89,15 @@ func NewDataMoverFromCluster(params map[string]interface{}, logger logrus.FieldL
 }
 
 func (this *DataMover) CopyToRepo(peID astrolabe.ProtectedEntityID) (astrolabe.ProtectedEntityID, error) {
+	backupRepository := builder.ForBackupRepository(utils.WithoutBackupRepository).Result()
```
Comment (deepakkinni) [inline pkg/dataMover/data_mover.go:92]: Could you change the call sites as well?, in upload controller check if the upload CR has a BR specified, if so call CopyToRepoWithBackupRepository

File: pkg/controller/upload_controller.go [hunk 7]
```diff
@@ -304,7 +306,25 @@ func (c *uploadController) processUpload(req *pluginv1api.Upload) error {
 		return errors.New(errMsg)
 	}
 
-	_, err = c.dataMover.CopyToRepo(peID)
+	if req.Spec.BackupRepositoryName != "" {
+		config, err := rest.InClusterConfig()
```
Comment (deepakkinni) [inline pkg/controller/upload_controller.go:310]: move this to a util method.

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/82/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/85/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/86/
